### PR TITLE
net : socket : make socket_tls depend on mbedtls

### PR DESCRIPTION
**Pull request by DenvE**
Socket_tls.c is depending on mbedtls.  If mbedtls is not configured, socket_tls gives compile issues

Signed-off-by: Wim Van Loocke denv8029@gmail.com

### CODE REVIEW

**Comment by jukkar** [comment]
The change looks ok, you need to resubmit the PR and fix the commit message. It is not enough to edit the PR description. Just fix the commit msg and force push the branch.
So the commit msg should look something like this:
```
net: socket: Add dependency to mbedtls

Socket_tls.c is depending on mbedtls. If mbedtls is not configured, socket_tls gives
compile issues

Signed-off-by: Wim Van Loocke denv8029@gmail.com
```

**Comment by jukkar** [comment]
Something went wrong here. Please do not do merge, but do rebase instead (so you should end up with one commit only), then force push you one commit branch do this.

File: subsys/net/lib/sockets/Kconfig [hunk 1]
```diff
@@ -48,6 +48,7 @@ config NET_SOCKETS_DNS_TIMEOUT
 
 config NET_SOCKETS_SOCKOPT_TLS
 	bool "Enable TCP TLS socket option support [EXPERIMENTAL]"
+	depends on MBEDTLS
```
Comment (rlubos) [inline subsys/net/lib/sockets/Kconfig]: Actually I'd make it: `select MBEDTLS if NET_NATIVE`

I saw that `NET_SOCKETS_SOCKOPT_TLS` option is also used in some offloaded engines (SimpleLink for instance) as well as in some application protocols (like LWM2M). mbedTLS is only needed for native implementation.

**Comment by DenvE** [comment]
Hi Robert, Hi Jukka,

I will do next steps:


- enter new problem report
- submit one commit with this info

"net: socket: Add dependency to mbedtls


Socket_tls.c is depending on mbedtls. If mbedtls is not configured,
socket_tls gives
compile issues


fixed : # ...

Signed-off-by: Wim Van Loocke denv8029@gmail.com"



with the solution proposed by Robert.

Hope it works then.

kr,

Wim

**Comment by jukkar** [comment]
> - enter new problem report

There is no need to create a new issue, the existing #21768 is enough

**Comment by DenvE** [comment]
Hi Robert,

if I make it *select MBEDTLS if NET_NATIVE*, it won't add mbedtls to the
configuration and it doesn't build.

I have read the documentation, but got confused as it is stated

The select statement is used to force one symbol to y whenever another
symbol is y.

In the .config is see

CONFIG_NET_NATIVE=y, so I expect CONFIG_MBEDTLS=y but got  # CONFIG_MBEDTLS
is not set

Any idea where it goes wrong?

envir :

samples/net/dhcpv4_client
west build -b stm32f746g_disco
branch zephyr master

Kr,

Wim

**Comment by rlubos** [comment]
Hi Wim,

Did you delete the build directory after modifying the Kconfig file? It's often the case that CMake keeps some stale configuration after the Kconfig/Cmake files are modified, so you need to give it a chance to re-create the project from scratch.

FYI, I've tested the configuration you've posted and it worked for me. I needed to enable `NET_SOCKETS_POSIX_NAMES` though, if you find this an issue it could be easily fixed by replacing `recv` call in `sockets_tls.c` file with `zsock_recv`.

**Comment by DenvE** [comment]
Hi Robert,

I have done that.( I always do that as building is not taking a lot of
time).

My issue is have seen was indeed related to recv, If I add
NET_SOCKETS_POSIX_NAMES=y,
indeed it is building.

Should sockets_tls.c be changed then to

#if defined(CONFIG_NET_SOCKETS_POSIX_NAMES)
if (recv(pfd->fd, NULL, 0, ZSOCK_MSG_DONTWAIT) < 0 &&
errno != EAGAIN) {
pfd->revents |= ZSOCK_POLLERR;
goto next;
}
#else
if (zsock_recv(pfd->fd, NULL, 0, ZSOCK_MSG_DONTWAIT) < 0 &&
errno != EAGAIN) {
pfd->revents |= ZSOCK_POLLERR;
goto next;
}
#endif


I guess it is better no to have "hidden" configs,.. so I you want I can
make issue and send a merge request.

kr,

Wim

**Comment by rlubos** [comment]
@DenvE There's no need to add extra complexity with `#if defined`. Simply replacing the existing `recv` call with `zsock_recv` is a safe operation - they're equivalent and the latter is always available. `recv` is more commonly used alias for `zsock_recv`, though it needs extra configs to be enabled (as it's a part of POSIX API).

IMO it's would be acceptable to have this change as well in this PR, as a separate commit. You 'd need to cleanup your commits in this PR though, I'd expect to see one commit adding mbedTLS dependency, and (optionally) one commit sorting-out the `recv` call. As @jukkar said, you need to rebase your branch towards master, not merge (or probably start fresh if you find this too complex).

**Comment by DenvE** [comment]
Hi Robert,

I have submitted the change for recv.

I got a mail  with this content
Identity/Emails issues

390e703144698ee011d7f2ef60d5423023bc37a6: author email (wimvanloocke
denv8029@gmail.com) needs to match one of the signed-off-by entries.
390e703144698ee011d7f2ef60d5423023bc37a6: author email (wimvanloocke
denv8029@gmail.com) does not follow the syntax: First Last .

I don't understand what it means First Last? and how to correct.


kr,


Wim

**Comment by jukkar** [comment]
There is now two PRs #21834 and #21892, are you now working on the latter?
In that latter PR, the commit messages are empty for the two commits, you need to fix that. Easiest is just do `git rebase -i HEAD~2` and edit the commit messages to be proper and force push.

**Comment by rlubos** [comment]
@DenvE As Jukka say, you need to sort out the commit message. Zephyr has quite strict requirements in terms of commit format. You need a proper title, body and signoff entry.

As for the signoff, don't do this manually. Git will add this for you if you specify the `--signoff` option when creating or amending a commit. Note, that you need to configure your username/email correctly in your local git setup, in order to get correct entries:
https://help.github.com/en/github/using-git/setting-your-username-in-git
https://help.github.com/en/github/setting-up-and-managing-your-github-user-account/setting-your-commit-email-address

**Comment by jukkar** [comment]
ping @DenvE What is your plan with this PR? It is basically ready, you just need to figure out your local git branch (to have only one commit there) and add proper signed-off reported by sanitychecker and we are done.

**Comment by jukkar** [comment]
Already fixed by #22359

**Pull request closed without merging**